src/kagglehub/gcs_upload.py

```diff
@@ -25,6 +22,18 @@
 REQUEST_TIMEOUT = 600
 
 
+class UploadDirectoryInfo:
+    def __init__(
+        self,
+        name: str,
+        files: Optional[List[str]] = None,
```

Comment:
Create a `class UploadFileInfo` and add a `serialize` method. See my comment about the `serialize` model for `UploadDirectoryInfo`.

Reply:
resolved offline. (Will keep files a List[str])